src/compute/src/sink/subscribe.rs

```diff
@@ -77,21 +79,27 @@ fn subscribe<G>(
     sinked_collection: Collection<G, Row, Diff>,
     sink_id: GlobalId,
     as_of: SinkAsOf,
+    up_to: Antichain<G::Timestamp>,
     subscribe_protocol_handle: Rc<RefCell<Option<SubscribeProtocol>>>,
 ) where
     G: Scope<Timestamp = Timestamp>,
 {
     let mut results = Vec::new();
+    let mut finished = false;
     sinked_collection
         .inner
         .sink(Pipeline, &format!("subscribe-{}", sink_id), move |input| {
+            if finished {
+                return;
+            }
```

Comment:
This can be problematic if the operator still receives data but doesn't drain its inputs. Before returning, it should drain all its inputs, otherwise it might cause the operator to be constantly rescheduled.